tools/yaml-templates/web-e2e-tests-job.yml

```diff
@@ -0,0 +1,73 @@
+# This template handles running E2E tests (including all necessary setup steps) from the host SDK repo for various environments.
+# It will run one E2E job for each testPrefixPattern selector in the testPrefixPatternGroups parameter.
+
+parameters:
+  - name: AppHostingSdk
+    default: none
+    type: string
+  - name: hostingEnvironmentType
+    type: string
+    values:
+      - 'electron'
+      - 'standardWeb'
+  - name: teamsJsReferenceType
+    type: string
+    values:
+      - 'npm'
+      - 'scriptTag'
+  - name: testPrefixPatternGroups
+    type: object
+    default: ['{[A-Z],[a-z],[0-9]}']
+
+jobs:
+  - ${{each testPrefixPattern in parameters.testPrefixPatternGroups}}:
+      - job: E2ETestsWeb_${{parameters.teamsJsReferenceType}}_${{ replace(replace(replace(replace(replace(replace(testPrefixPattern, '[', '_'), ']', '_' ), '{', '_'), '}', '_'), ',', '_'), '-', '_') }}
+        displayName: 'E2E Tests - Web - Via ${{parameters.teamsJsReferenceType}} - ${{parameters.hostingEnvironmentType}} Hosted - ${{testPrefixPattern}}'
+        pool:
+          name: Azure-Pipelines-1ESPT-ExDShared
+          image: 'ubuntu-2004'
+          os: linux
+        variables:
+          - name: envType
+            ${{ if eq(parameters.teamsJsReferenceType, 'npm') }}:
+              value: 'test'
+            ${{ else }}:
+              value: 'localScriptTag'
+        steps:
+          - template: build-app-host.yml
+            parameters:
+              appHostGitPath: AppHostingSdk
+
+          - task: CmdLine@2
+            displayName: 'Build Local Test App to Reference via Script Tag'
+            condition: and(succeeded(), eq('${{ parameters.teamsJsReferenceType }}', 'scriptTag'))
+            inputs:
+              script: |
+                pnpm build-test-app-local
+              workingDirectory: '$(ClientSdkProjectDirectory)'
+
+          - bash: >
+              node tools/cli/runAppsWithE2ETests.js
+              --useDataFromLocal=true
+              --testPrefixPattern "${{testPrefixPattern}}"
+              --envType=${{ variables.envType }}
+            displayName: 'Run web hosted E2E integration tests (${{testPrefixPattern}})'
+            condition: and(succeeded(), eq('${{ parameters.hostingEnvironmentType }}', 'standardWeb'))
+            workingDirectory: '$(AppHostingSdkProjectDirectory)'
+
+          - bash: |
+              # This sets up the server but runs no tests
+              node tools/cli/runAppsWithE2ETests.js --useDataFromLocal=true --testPrefixPattern "{[]}"
```

Comment:
The electron tests require the test host to be running before you run them, but there's no direct script to only setup the test host as needed right now, so I have to use the full "run e2e tests" script but provide it no tests to run (note the `{[]}` prefix pattern) which will cause it to setup the test host which can then be used by playwright below.

As a follow-up item, I will look to refactor the `runAppsWithE2ETests.js` in the other repo so that we can have a script that only sets up the test host that can be used here.